api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -301,6 +301,12 @@
      */
     int getId();
 
+    /**
+     * Return the number of windows before the next checkpoint including the current window.
+     * @return Number of windows from checkpoint, 1 if the checkpoint will be after the current window
+     */
+    int getWindowsFromCheckpoint();
```

Comment:
@PramodSSImmaneni I don't see how we can say this is a backward incompatible change just in theory. Is there a way to find out if none of the users implement this interface?

Also why are you going to use something like an attribute?